client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -51,6 +51,37 @@ interface IAnalysisWizard {
   isOpen: boolean;
 }
 
+const determineMode = (
+  applications: Application[]
+):
+  | "binary"
+  | "source-code-deps"
+  | "source-code"
+  | "binary-upload"
+  | undefined => {
+  // If only one application is selected
+  if (applications.length === 1) {
+    const { repository, binary } = applications[0];
+    // If the application has a repository or both repository and binary definitions, return "source-code-deps"
+    // If the application has only binary definitions, return "binary"
+    // If no definitions are present, return undefined
+    return repository || (repository && binary)
+      ? "source-code-deps"
+      : binary && binary !== ""
+      ? "binary"
+      : undefined;
```

Comment:
This looks good.  Just put it in a function, iterate through the applications array and get the mode for each application.  At the end either all of the modes are the same, or they are different.  Same = return that mode, different = return undefined.